open-api/typescript-sdk/fetch-client.ts

```diff
@@ -0,0 +1,2526 @@
+/**
+ * Immich
+ * 1.94.1
+ * DO NOT MODIFY - This file has been generated using oazapfts.
+ * See https://www.npmjs.com/package/oazapfts
+ */
+import * as Oazapfts from "oazapfts/lib/runtime";
+import * as QS from "oazapfts/lib/runtime/query";
+export const defaults: Oazapfts.Defaults<Oazapfts.CustomHeaders> = {
+    headers: {},
```

Comment:
This object exists for stuff like setting up auth and what not. I'm pretty sure we can use this instead of keeping the ImmichApi class around.

Reply:
The `ImmichApi` class is actually pretty helpful for testing as we can mock the functions on it. If the code directly imported the functions we'd need to figure out some solution for mocking them

I also was a little wary of using the `defaults` object because it's a global and I didn't know how well that would work if we ran multiple tests in parallel that ended up stomping on each other's options

Reply:
while true, I thought the main reason was to help with code splitting. if the web has to use a class to enable easier testing than none of this will really be beneficial in that regard.

Reply:
yeah, I'd take a different approach in the web project. here it was fine to stick everything in a single class because we're just bundling to a single distributable